Src/AutoFixture.xUnit.net/FreezeAttribute.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Reflection;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.Xunit
+{
+    /// <summary>
+    /// An attribute that can be applied to parameters in an <see cref="AutoDataAttribute"/>-driven
+    /// Theory to indicate that the parameter value should be frozen and used to satisfy any requests
+    /// made through the same <see cref="IFixture"/> and that match a set of criteria.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+    public sealed class FreezeAttribute : CustomizeAttribute
```

Comment:
I'm not too happy combining `As` and `By`... Although I haven't fully thought it through in this particular case, this sounds like a recipe for a really confusing API.

My assumption about this feature is that most (90%) of the use cases for the `As` property could be replaced by `Matching.ImplementedInterfaces`, or am I mistaken?

There may be some rare situations where one interface derives from another interface, and I'm not sure `Type.GetInterfaces()` returns those, but apart from that situation, I wonder if it wouldn't work out if we deprecate the `As` property in favour of the `By` property.

What do you think?